controllers/targetallocator_controller.go

```diff
@@ -55,17 +63,42 @@ type TargetAllocatorReconcilerParams struct {
 	Config   config.Config
 }
 
-func (r *TargetAllocatorReconciler) getParams(instance v1alpha1.TargetAllocator) targetallocator.Params {
+func (r *TargetAllocatorReconciler) getParams(ctx context.Context, instance v1alpha1.TargetAllocator) (targetallocator.Params, error) {
+	collector, err := r.getCollector(ctx, instance)
+	if err != nil {
+		return targetallocator.Params{}, err
+	}
 	p := targetallocator.Params{
 		Config:          r.config,
 		Client:          r.Client,
 		Log:             r.log,
 		Scheme:          r.scheme,
 		Recorder:        r.recorder,
 		TargetAllocator: instance,
+		Collector:       collector,
+	}
+
+	return p, nil
+}
+
+func (r *TargetAllocatorReconciler) getCollector(ctx context.Context, instance v1alpha1.TargetAllocator) (*v1beta1.OpenTelemetryCollector, error) {
+	var collector v1beta1.OpenTelemetryCollector
+	ownerReferences := instance.GetOwnerReferences()
+	collectorIndex := slices.IndexFunc(ownerReferences, func(reference metav1.OwnerReference) bool {
+		return reference.Kind == "OpenTelemetryCollector"
+	})
+	if collectorIndex != -1 {
```

Comment:
in the future this is where we'll do the label selector get i assume?

Reply:
Yep, that's coming in a follow-up.